### PR TITLE
JAMES-2586 JMAP Guice bindings modules to pg-app

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import org.apache.james.vacation.api.VacationRepository;
+import org.apache.james.vacation.api.VacationService;
+import org.apache.james.vacation.memory.MemoryNotificationRegistry;
+import org.apache.james.vacation.memory.MemoryVacationRepository;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Scopes;
+import com.google.inject.name.Names;
+
+public class PostgresJmapModule extends AbstractModule {
```
Comment (quantranhong1999) [inline server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java:48]: Should not be directly in postgres-app module, and same place as `PostgresDataJmapModule` IMO. 

BTW `PostgresDataJmapModule` and `PostgresJmapModule` make me confuse. I am not sure what is the diff between those two. Any chance to merge them, or split them with better name?

**Review by quantranhong1999 — COMMENTED** [review]
Q: Any plan to plug other modules `JmapEventBusModule`, `JMAPListenerModule` ... in this PR?

But could split this as a smaller PR to promote postgres-webadmin-IT ticketIMO.

**Comment by hungphan227** [comment]
> Q: Any plan to plug other modules `JmapEventBusModule`, `JMAPListenerModule` ... in this PR?
> 
> But could split this as a smaller PR to promote postgres-webadmin-IT ticket IMO.

What do you mean? I have added them in PostgresJamesServerMain

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java (hunk 1 above)
Comment (hungphan227) [reply server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java:48]: We discussed this previously in other PR. Until now, I am still not sure if I should put these bindings in a separated module as I did or put them in PostgresMailboxModule and PostgresDataModule?

**Comment by quantranhong1999** [comment]
> What do you mean? I have added them in PostgresJamesServerMain

Sorry, I missed that. Great ^^.

**Review by Arsnael — APPROVED** [review]

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java (hunk 1 above)
Comment (Arsnael) [reply server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java:48]: Link to that conversation then please? I find weird the two modules as well actually now that I'm thinking about it again

**Review by vttranlina — APPROVED** [review]

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJamesServerMain.java [hunk 2]
```diff
@@ -94,20 +97,26 @@ public class PostgresJamesServerMain implements JamesServerMain {
...
         new TikaMailboxModule());
 
+    public static final Module JMAP = Modules.combine(
+        new PostgresJmapModule(),
+        new JmapEventBusModule(),
+        new PostgresDataJmapModule(),
+        new MemoryPushSubscriptionModule(),
+        new JMAPServerModule(),
+        new JmapTasksModule(),
+        new JMAPListenerModule());
```
Comment (Arsnael) [inline server/apps/postgres-app/src/main/java/org/apache/james/PostgresJamesServerMain.java:116]: What about the other `JMAPEventBusModule` ?

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java (hunk 1 above)
Comment (hungphan227) [reply server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java:48]: https://github.com/apache/james-project/pull/1909#discussion_r1446853591

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJamesServerMain.java (hunk 2 above)
Comment (hungphan227) [reply server/apps/postgres-app/src/main/java/org/apache/james/PostgresJamesServerMain.java]: added

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java (hunk 1 above)
Comment (Arsnael) [reply server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java:48]: I'm ok to keep it like that for now. Can always refactor little things like this later.

**Review by Arsnael — APPROVED** [review]

File: server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java (hunk 1 above)
Comment (hungphan227) [reply server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java]: moved to PostgresAdditionalDataJmapModule in postgres-common

**Comment by Arsnael** [comment]
I feel I'm confused again... I don't understand the last commit? Pretty sure also you gonna get checkstyle issues on that :) 

I think it was fine before that last fixup

**Comment by hungphan227** [comment]
> I feel I'm confused again... I don't understand the last commit? Pretty sure also you gonna get checkstyle issues on that :)
> 
> I think it was fine before that last fixup

I moved bindings to postgres-common because I see that we do not put bindings in postgres-app

**Comment by hungphan227** [comment]
org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*database system is ready to accept connections.*\s'

**Comment by Arsnael** [comment]
> I moved bindings to postgres-common because I see that we do not put bindings in postgres-app

It makes you add mailbox memory dependency to postgres-common though, which I am more bothered by TBH

**Comment by hungphan227** [comment]
> > I moved bindings to postgres-common because I see that we do not put bindings in postgres-app
> 
> It makes you add mailbox memory dependency to postgres-common though, which I am more bothered by TBH

:(( Do you want me to move them back to postgres-app?

**Comment by Arsnael** [comment]
Me I'm ok with this PR without the last commit.

@quantranhong1999 WDYT? As you initially raised the debate :)

**Comment by quantranhong1999** [comment]
> @quantranhong1999 WDYT? As you initially raised the debate :)

Generally, I am easy on this. As we can always refactor later as discussed during the daily.

**Comment by Arsnael** [comment]
i mean I prefer not having too much memory deps being injected everywhere then. Can you please repush this PR without the last commit @hungphan227 ? Thanks and sorry!

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**